vignettes/stages.Rmd

```diff
@@ -0,0 +1,97 @@
+#> Please use `tibble()` instead.
+#> This warning is displayed once every 8 hours.
+#> Call `lifecycle::last_warnings()` to see where this warning was generated. 
+```
+
+The deprecating warning tells you when the function was deprecated (in tibble 1.1.0 released in 2016), and what to use instead (`tibble()`).
+To avoid being too annoying, deprecation messages will only appear once per session, and you can find out exactly where they come from by calling `lifecycle::last_warning()`.
+`vignette("user")` provides more advice on handling deprecation warnings in your code.
+
+Particularly important functions may go through two additional stages of deprecation:
+
+-   **Soft deprecated** comes before deprecated.
+    It's a gentler form of deprecation designed to prevent new uses of a function and encourage package developers to move away from it.
+    Soft deprecated allows a package to change its interface to encourage package developers to update their code before users are forced to change.
+
+-   **Defunct** comes after deprecated.
+    In most cases, a deprecated function will eventually just be deleted.
+    For very important functions, we'll instead make the function defunct, which means that function continues to exist but the deprecation warning turns into an error.
+
+## Superseded
+
+A softer alternative to deprecation is superseded.
+A ![superseded](figures/lifecycle-superseded.svg)[^1] function has a known better alternative, but the function itself is not going away
+. A superseded function will not emit a warning (since there's no risk if you keep using it), but the documentation will tell you what we recommend instead
+.
+
+[^1]: This stage was previously called retired.
+
+Superseded functions will not receive new features, but will receive any critical bug fixes needed to keep it working.
+In some ways a superseded function is actually safer than a stable function because it's guaranteed never to change (for better or for worse).
+
+## Experimental
+
+Some functions are released in an ![experimental](figures/lifecycle-experimental.svg){style="vertical-align:middle"} stage.
+Experimental functions are made available so people can try them out and provide feedback, but come with no promises for long term stability.
+In particular, the author reserves the right to make breaking changes without a deprecation cycle.
+That said, there is some interaction between popularity and stability.
+Breaking an popular function, even if clearly labelled as experimental, is likely to cause widespread pain so we'll generally try to avoid it.
+
+In general, you can assume any package with version number less than 1.0.0 is at least somewhat experimental, and it may have major changes in its future.
+The most experimental packages only exist on GitHub.
+If you're using a non-CRAN package you should plan for an active relationship: when the package changes, you need to be prepared to update your code.
+
+## Questioning
```

Comment:
I wonder if we can just mark these functions with `@keyword internal` temporarily? Then we don't need this additional lifecycle stage which I don't find very helpful. Like "maturing", we could just mention the questioning status in the documentation if needed.

Reply:
I think we've used questioning in a couple of places so we need to keep this on the vignette, but I'll rewrite just to say that it's a stage we no longer use.

Reply:
You could say that "questioning" is "superseded". 🤣